modules/auxiliary/scanner/smb/smb_ms17_010.rb

```diff
@@ -93,14 +94,16 @@ def run_host(ip)
           case dcerpc_getarch
           when ARCH_X86
             os << ' x86 (32-bit)'
+            details[:arch] = ARCH_X86
           when ARCH_X64
             os << ' x64 (64-bit)'
+            details[:arch] = ARCH_X64
           end
```

Comment:
These details should be reported into the database along with the host information.

Reply:
Fixed in commit 3fdeb1933d69f583a11a5db6ab88e6b50b630a26.

Example:
```
msf6 auxiliary(scanner/smb/smb_ms17_010) > hosts -c address,name,arch,os_family,os_name,os_flavor

Hosts
=====

address  name  arch  os_family  os_name  os_flavor
-------  ----  ----  ---------  -------  ---------

msf6 auxiliary(scanner/smb/smb_ms17_010) > rerun
[*] Reloading module...

[+] 192.168.159.38:445    - Host is likely VULNERABLE to MS17-010! - Windows 7 Professional 7601 Service Pack 1 x64 (64-bit)
[*] 192.168.159.38:445    - Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/smb/smb_ms17_010) > hosts -c address,name,arch,os_family,os_name,os_flavor

Hosts
=====

address         name             arch  os_family  os_name  os_flavor
-------         ----             ----  ---------  -------  ---------
192.168.159.38  WIN-9NSI4A6AIHJ  x64   Windows             Professional

msf6 auxiliary(scanner/smb/smb_ms17_010) >
```

Unfortunately I can't figure out how to accurately report the os_name from Recog.  When I add `os_name: fp_match['os.product'],` to the `report_host` call the output is modified to the following. Notice the "7Professional". I tracked this down to this block [here](https://github.com/rapid7/metasploit-framework/blob/07cbe426e2f80f523422487c83641ac9ee7b7971/lib/msf/core/db_manager/host.rb#L232-L242) and can't figure out how to bypass it. Any thoughts @acammack-r7?
```
msf6 auxiliary(scanner/smb/smb_ms17_010) > hosts -c address,arch,created_at,mac,name,os_family,os_flavor,os_lang,os_name,os_sp,purpose,service_count,state,updated_at,virtual_host,vuln_count

Hosts
=====

address         arch  created_at               mac  name  os_family  os_flavor      os_lang  os_name  os_sp  purpose  service_count  state  updated_at               virtual_host  vuln_count
-------         ----  ----------               ---  ----  ---------  ---------      -------  -------  -----  -------  -------------  -----  ----------               ------------  ----------
192.168.159.38  x64   2020-10-22 21:28:49 UTC             Windows    7Professional           Windows                  1              alive  2020-10-22 21:28:49 UTC                1

msf6 auxiliary(scanner/smb/smb_ms17_010) > %                                                                                  
```
Ideally I'd be able to report the same results that the `smb_version` module gives:
```
address         arch  created_at               mac  name             os_family  os_flavor     os_lang  os_name    os_sp  purpose  service_count  state  updated_at               virtual_host  vuln_count
-------         ----  ----------               ---  ----             ---------  ---------     -------  -------    -----  -------  -------------  -----  ----------               ------------  ----------
192.168.159.38        2020-10-22 21:27:58 UTC       WIN-9NSI4A6AIHJ  Windows    Professional           Windows 7  SP1    client   1              alive  2020-10-22 21:27:58 UTC                1
```

Reply:
Hm, that looks to be behavior from the targeting refactor in https://github.com/rapid7/metasploit-framework/pull/7766. I think sensible changes to `report_host` to do the right thing with `:os_flavor` would be fine. No automation or matching use that field AFAIK.

Reply:
Okay well I hope by "do the right thing" you meant to remove the alterations to the `os_name` and `os_flavor` keys altogether because that's what I did in 291c883db8d3470a2ee1d6c77754408300006f0e. If I misunderstood, I'm happy to change it.

Reply:
I think automatically populating it for Windows OSes that don't specify it manually would be alright, since very few things do set it manually from what I can see. Either way, you probably have more insight on what would be helpful to users/module writers.